### ISSUE TITLE
Does not support more than 10 players with the same name on 16 char usernames

### ISSUE THREAD

**Issue by spaceFountain** [Civcraft/NameLayer#67]
Currently the procedure for determining a name for a new player only takes off the last character from a user name no matter the length of the string that must be appended before adding the new string. 

There are also problems with a possibility of two players getting the same name. It can be seen like this. Player with username "aa" logs and then changes their name to bb. A new player player with username "aa" logs on and is given the name "aa1". Player with username "aa1" logs on. What happens? If the last char is always cut off as it looks like might be the case (I don't know how to properly read the scripting language that is used with SQL) the same problem can be triggered with slightly different names

**Comment by suirad** [comment]
This seems very situational and extremely unlikely. I dont think this should be critical.

**Comment by ttk2** [comment]
Probably not.

**Comment by erocs** [comment]
It also creates names greater then the max supported 16 character length by appending the number to 16 character names.

Bump.

**Comment by tyrothalos** [comment]
Rather than try and assign a player a name that hasn't already been taken, couldn't it just have players choose a name the first time they log on? 
If the name they choose is taken or invalid, make them keep choosing names until they find one that isn't taken. Until then prevent them from being able to play, similar to how they can't do anything until they agree to civcraft's rules.

**Comment by Maxopoly** [comment]
If we make sure the name is taken neither on civ, nor on minecraft in general, that could work out. Alternatively we could also just fix the naming scheme, looking at this issue that shouldnt be too hard.

**Comment by tyrothalos** [comment]
Does it matter if the name is already taken on minecraft in general? If someone takes the name of someone who doesn't play civcraft and then the non-civcrafter logs in one day, they just need to choose a different civcraft name. 

Anyhow, what do you have in mind for an alternative? The easiest and most reliable ways that come to my mind all have the potential of turning names into gibberish.

**Comment by Maxopoly** [comment]
Yes it does matter, because a lot of people want to get "good" names, but they can't get those, because the name is taken on Mojangs side. If we allowed them to chose a name on civ, many people would actively get a new account with the name of an existing account on civ, just so they could chose a new name here.

I'll quickly make something that should imo adress this problem

**Comment by tyrothalos** [comment]
Actually I was thinking that everyone that logs in gets to choose a new name, not just the ones with name conflicts, but it looks like almost have this sorted out so forget what I said.
